cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -969,7 +969,8 @@ private double computeBalanceUpperThreshold(double clusterUtilization, Optimizat
    * @return The utilization lower threshold in percent for the {@link #resource()}
    */
   private double computeBalanceLowerThreshold(double clusterUtilization, OptimizationOptions optimizationOptions) {
-    return clusterUtilization * Math.max(0, (1 - balancePercentageWithMargin(optimizationOptions)));
+    return Math.max(clusterUtilization * Math.max(0, (1 - balancePercentageWithMargin(optimizationOptions))),
+                    _balancingConstraint.lowUtilizationThreshold(resource()));
```

Comment:
This does not seem right.

Background: (1) Resource distribution goals try to make the utilization over each broker between some _lower_ and _upper_ threshold. (2) The purpose of the `lowUtilizationThreshold` config is to avoid triggering actions if _all brokers_ are under some configured threshold. In a way, it makes CC take no action if _all brokers_ are under a certain utilization. (please see the relevant documentation on `AnalyzerConfig`)

However, this PR pushes the lower threshold potentially to a higher value. This potentially will require brokers with a low utilization to try to receive more load from the other brokers.

What we need to do is to
1. identify if all the brokers are below the low utilization threshold 
2. If step 1 is true, then set the lower threshold to `0` and upper threshold to the config value (i.e. `_balancingConstraint.lowUtilizationThreshold(resource())`).

This PR also needs to update how `_numBalancedBrokersByResource` is computed in `ClusterModelStats`.

Also let's add some test cases to verify the functionality.